server/src/main/java/org/opensearch/action/search/DeletePitResponse.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.action.ActionResponse;
+import org.opensearch.common.ParseField;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ConstructingObjectParser;
+import org.opensearch.common.xcontent.StatusToXContentObject;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.opensearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.opensearch.rest.RestStatus.NOT_FOUND;
+import static org.opensearch.rest.RestStatus.OK;
+
+/**
+ * Response class for delete pits flow which clears the point in time search contexts
+ */
+public class DeletePitResponse extends ActionResponse implements StatusToXContentObject {
+
+    private final List<DeletePitInfo> deletePitResults;
+
+    public DeletePitResponse(List<DeletePitInfo> deletePitResults) {
+        this.deletePitResults = deletePitResults;
+    }
+
+    public DeletePitResponse(StreamInput in) throws IOException {
+        super(in);
+        int size = in.readVInt();
+        deletePitResults = new ArrayList<>();
+        for (int i = 0; i < size; i++) {
+            deletePitResults.add(new DeletePitInfo(in));
+        }
+
+    }
+
+    public List<DeletePitInfo> getDeletePitResults() {
+        return deletePitResults;
+    }
+
+    /**
+     * @return Whether the attempt to delete PIT was successful.
+     */
+
+    @Override
+    public RestStatus status() {
+        if (deletePitResults.isEmpty()) return NOT_FOUND;
```

Comment:
What would be the status if PIT ID exists but we are not able to delete it?

Reply:
Succeeded would be false for the PIT ID deletion failures as given in PR description. Status is still 200.